fabtests/component/dmabuf-rdma/fi-rdmabw-xe.c

```
@@ -1205,7 +1224,13 @@ int main(int argc, char *argv[])
 			use_sync_ofi = 1;
 			break;
 		case 'S':
-			msg_size = atoi(optarg);
+			msg_size = parse_size(optarg);
+			max_size = msg_size;
+			proxy_block = msg_size;
+			break;
+		case 'M':
+			max_size = parse_size(optarg);
+			proxy_block = max_size;
```

I think all RDMA protocols have a 2GB message limit

---

I don't think we need to enforce that limit here. Just let the run fail if the specified size is too large.

---

The memory registration is limited as well, however.

---

It should also fail by itself. 

However, in a few places int the test, `int` is used for the size. That actually can overflow and cut off the test at 2GB. Let me make the change to see how the test behavior for really large sizes.

---

It's not obvious that changing the batch size can result in memory registration overflow.  I'd add a check with an error message to make the problem obvious, versus the user getting an error later for what looks like an unrelated problem.

---

I ran some tests. It appears that registration itself is fine, but the RDMA fails after 1GB.  I tried `-b 64 -M 1G -n 128` and it worked fine which means the entire 64G memory is registered & accessed correctly if the transfer size doesn't exceed 1GB. However, the run with `-b 1 -M 1073741825 -n 1` failed immediately with I/O error.

---

We could check if `(max_size * batch)` overflow `size_t` though.